### PR TITLE
PT-4027: Rework the comment filter bar into orthogonal dropdowns

### PR DESCRIPTION
**Pull request by rolfheij-sil**
Reworks the comments-panel filter toolbar from a single grab-bag dropdown (which mixed unrelated axes into a handful of presets) into a row of **orthogonal filter dropdowns** that AND together, alongside the existing scope dropdown. This delivers PT-4027's "unresolved conflicts" view (and more) by composition.

### The filter axes

Each dropdown defaults to **all** (so the default, unfiltered view is unchanged):

| Dropdown | Options | Selector field |
| --- | --- | --- |
| **Resolved** | All statuses / Unresolved / Resolved | `isResolved` |
| **Read** | All / Unread / Read | `isRead` |
| **Type** | All types / Conflicts / Comments | `type` (`Conflict` / `Normal`) |
| **Assignment** | Anyone / Assigned to me / Team | `assignedTo` |
| **Scope** *(unchanged)* | All books / Current chapter | `scriptureRanges` |

PT9's notes-filter presets are reachable by composing axes — e.g. **PT-4027's unresolved-conflicts view** = Type: Conflicts + Resolved: Unresolved; the old "unresolved assigned to me" = Resolved: Unresolved + Assignment: Me; "unread and unresolved" = Read: Unread + Resolved: Unresolved. This mirrors the axes of PT9's **"Define Filter"** dialog (PT9's own toolbar is likewise a single preset dropdown — this is a UX improvement over it).

### Data layer

One new selector field, `isResolved?: boolean` (mirrors the existing `isRead`) — a nullable `bool? IsResolved` on `CommentThreadSelector`, parsed in the converter, applied as one `Where` clause in `GetCommentThreads`. `status: 'Todo'` was deliberately not used — it would hide `Done`/`Unspecified` threads. **Every other axis maps to a field that already existed** (`isRead` / `type` / `assignedTo`), so the toolbar rework is otherwise pure frontend.

The filter constants and the selector-building logic live in a dependency-free `comment-list-filters.model.ts` so the mapping is unit-tested; the panel renders the dropdowns via a small generic `FilterDropdown` helper.

### Deliberately out of scope (kept small; flag if wanted)

- **Unassigned** as an assignment option — would need one small provider change (empty `assignedTo` currently means "no filter"); dropped for now, so Assignment is All / Me / Team.
- **Tag** filtering (To Do / user tags), **date** filters (PT9's "Recently created/resolved"), other PT9 user modes (Created By / Commented By), and **GCN** review states — all larger features, not attempted.

### Behavior note (intentional PT9 parity)

"Unresolved" keys purely on **thread status**, exactly like PT9. A conflict that was resolved (text applied) and then had its thread reopened (status → not-`Resolved`) reappears under Resolved: Unresolved — PT9 has no separate conflict-resolved datum. Deliberate parity, not a bug.

### Tests

- **C#**: converter parse for `isResolved` (3); provider filtering — `isResolved` incl. the `Done`-status edge and unset = no-filter (3); `type: Conflict` (both queries) and `type: Normal` returns only regular threads (verifies the "Comments" filter, since regular notes default to `NoteType.Normal`).
- **Filter model (vitest)**: each axis maps to its clause, axes AND together (incl. the unresolved-conflicts view), and compose with the current-chapter scope.

Jira: [PT-4027](https://paratextstudio.atlassian.net/browse/PT-4027) (sub-task of PT-4024). Independent, branched off `main`.

AI-assisted — [session](https://claude.ai/code/session_018ZurpFvP3QbVr71jzdtwKA)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_018ZurpFvP3QbVr71jzdtwKA

[PT-4027]: https://paratextstudio.atlassian.net/browse/PT-4027?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/paranext/paranext-core/2508)

---

## Automated review summary

# Code Review Summary

**Branch**: pt-4027-conflict-filters

**Base**: origin/main

**Date**: 2026-07-08

**Review model**: Claude Opus 4.8

**Files changed**: 12

## Overview

PT-4027 reworks the `legacy-comment-manager` filter bar from a single preset dropdown into
four orthogonal filter axes — resolved status, read status, note type, and assignment — that
AND together, plus the pre-existing scope axis. The per-axis mapping was extracted into a new
React/DOM-free `comment-list-filters.model.ts` (with unit tests) whose `buildCommentThreadSelector`
composes the axes into a `LegacyCommentThreadSelector`; the old Paratext 9 presets ("unresolved
conflicts", "unresolved assigned to me", etc.) are now reached by combining axes. On the C# side, a
new nullable `IsResolved` filter was added to `CommentThreadSelector` (+ its JSON converter and the
`ParatextProjectDataProvider` LINQ predicate) so the frontend can request only unresolved (or only
resolved) threads server-side — the stated PT-4027 "unresolved conflicts" goal.

This is a clean, well-tested change. Four `/review-paratext` passes (API/correctness, style/patterns,
coverage/compliance, UX) and a deeper `/code-review high` correctness pass independently found **no
critical and no important issues**. The `isResolved` tri-state was traced end-to-end (TS `isResolved?`
↔ JSON `isResolved` ↔ C# `bool?`; predicate `(Status == Resolved) == isResolved`, with `null`
skipping the filter), the axes compose correctly, dead preset code is fully removed (grep-clean), the
React memo deps are complete, and both sides carry thorough tests including the "unresolved conflicts"
path end-to-end.

## API Changes

- `extensions/src/legacy-comment-manager/src/types/legacy-comment-manager.d.ts`: Added optional
  `isResolved?: boolean` to the exported `LegacyCommentThreadSelector` type. Purely additive; unset =
  no filtering, so existing behavior is preserved. Its C# counterpart is `CommentThreadSelector.IsResolved`
  (`bool?`), kept in sync across the TS ↔ JSON ↔ C# contract. No other public API surfaces changed
  (`lib/platform-bible-react`, `lib/platform-bible-utils`, `lib/papi-dts/papi.d.ts` untouched).

### Critical — Must address before merge

None.

### Important — Should address before merge

None.

### Minor — Consider

- [ ] ~~The "Assigned to me" axis is a silent no-op during the registration-data race: `currentUserName`
  starts `''` and loads async, so selecting it in the first render frames yields `{assignedTo:''}`, the C#
  guard skips the clause, and all threads show briefly before it self-corrects.~~
  _(Author: leave it for now — pre-existing behavior carried over from the old `FILTER_*_ASSIGNED` presets;
  very transient and self-correcting via the `useMemo` dep.)_ (`comment-list-filters.model.ts:148`,
  `comment-list.web-view.tsx:69`, `ParatextProjectDataProvider.cs:409`)
- [x] **`isResolved` overlaps the existing `status` axis** — `isResolved:true` ≡ `status:'Resolved'`, and a
  contradictory combination (e.g. `status:'Todo'` + `isResolved:true`) silently returns zero results.
  _(fixed during review: added a cross-reference TSDoc note on both `status` and `isResolved` in
  `legacy-comment-manager.d.ts` explaining they both filter thread status and should not be combined.)_
- [x] **`noFiltersActive` re-encoded the "each axis at 'all' = no filter" knowledge inline**, duplicating
  the model's axis defaults.
  _(fixed during review: added an exported `areCommentFiltersAtDefault(filters)` helper to the model —
  co-locating the axis-default knowledge with the axis definitions — and `comment-list.component.tsx` now
  uses it.)_
- [x] **The Storybook `filterThreads` hardcoded the `'Team'` sentinel** as a magic string while the model
  defined it privately.
  _(fixed during review: exported `TEAM_ASSIGNED_USER` from the model and reused it in the story, removing
  the magic-string drift risk.)_
- [x] **Only `isTypeFilter` of the five axis type-guards had a direct unit test.**
  _(fixed during review: replaced the single-guard test with a parametrized `describe.each` covering all
  five guards — `isScopeFilter`, `isResolvedFilter`, `isReadFilter`, `isTypeFilter`, `isAssignmentFilter` —
  plus a test for the new `areCommentFiltersAtDefault` helper. Model test count went from 6 cases to 17.)_
- [x] **The assignment "all" label `"Assigned to anyone"` broke the "All X" pattern** used by the other
  axes and misread as "has some assignee" (it actually includes unassigned threads).
  _(fixed during review: renamed to `"All assignments"` / `"Todas las asignaciones"` in en + es.)_
- [x] **The resolved-axis default `"All statuses"` was ambiguous next to the read-axis `"All read statuses"`.**
  _(fixed during review: renamed to `"All resolved statuses"` / `"Todos los estados de resolución"` in en +
  es, making the two status dropdowns parallel and self-describing.)_
- [ ] ~~`"Assigned to me"` uses first person, deviating from the second-person vocabulary guideline.~~
  _(Reviewer recommended leaving it — "Assigned to me" is an established filter idiom (GitHub/Jira) and
  "Assigned to you" reads worse; carried over from the prior labels. Author delegated ("fix as you see fit,
  low-risk only"); left unchanged deliberately.)_
- [ ] ~~The filter toolbar grew from two dropdowns to five (`min-w-32` each) and wraps to ~3 rows at the
  documented 300px webview min-width — at, but not over, the documented 3-row maximum.~~
  _(Left as a manual-QA note rather than a code change: the toolbar correctly uses `flex-wrap` (not
  control-hiding), and this is a rendering check best done visually in the narrow layout on long-label
  locales such as Spanish. No code defect.)_

#### Shared Regions Modified

None. No `#region shared with` markers exist in any of the 12 changed files.

#### Extension Config Changes

None. No `package.json`, `tsconfig*`, `webpack`, `.eslintrc*`, or CI/build files were changed. The only
non-source file touched is `contributions/localizedStrings.json`, an extension-specific data file — no
propagation needed.

## Positive Observations

- Public API change is purely additive and backward-compatible; the cross-process contract (TS `isResolved?`
  ↔ JSON `isResolved` ↔ C# `bool?`) is kept in sync, with `null`/unset = "no filtering" consistent on both
  sides.
- TSDoc and the C# `<summary>` carry matching, carefully-worded semantics (thread-status "resolved", not
  conflict-specific state; `false` matches Todo/Done/Unspecified, mirroring PT9 `StatusFilter(Unresolved)`).
- Good separation: the filter mapping was extracted into a React/DOM-free `comment-list-filters.model.ts`
  (following the extension's existing `*.model.ts` pattern), making `buildCommentThreadSelector` unit-testable.
- Strong reuse/DRY win: a generic `FilterDropdown<T>` collapses five near-identical `Select` blocks into one
  type-safe, self-narrowing component driven by per-axis configs.
- The C# `IsResolved` addition mirrors the existing `IsRead` sibling pattern exactly across all three touch
  points (converter constant + parse block, selector property, PDP LINQ filter).
- Localization hygiene: obsolete keys were retired and fresh axis-specific keys introduced (rather than
  editing existing keys' meanings); both en and es updated together; no dangling references to removed
  symbols/keys anywhere in the repo (grep-confirmed).
- Thorough tests on both sides: C# converter deserialization (true/false/absent→null), provider scenarios
  including "unresolved conflicts" end-to-end and the `IsResolved=false` includes-Done edge case; TS model
  tests for per-axis mapping and composition.
- React wiring is correct: `useMemo` deps are complete and `setFilters({...filters, x})` always yields a new
  reference, so the query re-runs on every filter change — no stale closures.

## Interview Notes

- **Purpose** (author-confirmed via PR title): rework the comment filter bar into orthogonal dropdown axes
  (Resolved / Read / Type / Assignment) and add a server-side `isResolved` thread-status filter enabling the
  "unresolved conflicts" view.
- **Finding #1 (assigned-to-me race)**: author chose to leave it — it is pre-existing behavior re-exposed as a
  standalone axis, transient, and self-correcting. Recorded as an accepted known limitation, not an oversight.
- **Finding #2 (status/isResolved overlap)**: author asked for the clarifying TSDoc note; added.
- **Remaining minors**: author delegated ("fix as you see fit; only if easy/cheap and low-risk"). Reviewer
  applied #3–#7 (all low-risk) and deliberately left #8 (established idiom) and #9 (a visual-QA check, not a
  code change) with rationale above.
- No areas were deferred to AI or left unexplained; the author engaged directly on the two correctness findings.

## In-Review Quality Check

Changes were made during review (findings #2–#7), touching six files, all within the `legacy-comment-manager`
extension (no `lib/` changes, so no library rebuild needed):

- **Tests**: `comment-list-filters.model.test.ts` — **17/17 pass** (was 6 cases; +11 from the parametrized
  guard tests and the `areCommentFiltersAtDefault` tests), run via the extension's vitest config.
- **Lint**: `eslint` on all five changed TS/TSX files — **clean (exit 0)**.
- **Format**: `prettier --check` on the changed files — one auto-format applied (reflowed a TSDoc line in
  `legacy-comment-manager.d.ts`); all files now clean.
- **Typecheck caveat**: a full project `tsc` could not be run cleanly *inside this isolated review worktree*.
  The worktree's `node_modules` is a symlink to the main checkout, whose npm-workspace self-link resolves
  `legacy-comment-manager` to the **other** checkout (currently on `pt-4030`, which predates the `isResolved`
  field). That produces spurious `TS2339 Property 'isResolved' does not exist` and `TS6200` duplicate-module
  errors — confirmed to be an environmental cross-checkout artifact (main's `.d.ts` has zero `isResolved`),
  **not** a code defect and **not** introduced by the in-review edits. The in-review changes add no dependency
  on `isResolved` and are type-trivial. **Recommendation**: run `npm run typecheck` in a normal checkout (or
  rely on CI) as the final pre-merge gate.

## Suggested Review Focus

- [ ] `isResolved` thread-status semantics vs the existing `status` axis — confirm the intended behavior of
  `isResolved:false` (Todo/Done/Unspecified) matches PT9 and that the new TSDoc note is sufficient guidance
  for future callers.
- [ ] The "Assigned to me" registration-data race (finding #1, accepted as-is) — confirm the transient
  "shows all threads" flash is acceptable UX, or whether the axis should hold until `currentUserName` loads.
- [ ] New microcopy: verify the renamed labels ("All assignments", "All resolved statuses") read well in the
  live dropdowns and that the Spanish translations are natural.
- [ ] Narrow-layout rendering (finding #9): confirm the five-dropdown toolbar does not spill past three rows
  at the 300px webview min-width on long-label locales.
- [ ] Full `npm run typecheck` in a standard checkout (see typecheck caveat above).

AI-assisted review — [session](https://claude.ai/code/session_015hfnmegSuB5vVmfNCvAtqk)

### CODE REVIEW

**Comment by lyonsil** [comment]
Regarding the review focus items, I think #4 below is the most important to consider. That is, this design diverges from PT9 in a meaningful way that changes UX. If others are already on board, OK, but I worry about design drift in general as it can become a time sink.

1. isResolved thread-status semantics vs the existing status axis — confirm the intended behavior of isResolved:false (Todo/Done/Unspecified) matches PT9 and that the new TSDoc note is sufficient guidance for future callers.
- Did you review this with Sebastian, Tim, John, or someone else from P9?

2. The "Assigned to me" registration-data race (finding #1, accepted as-is) — confirm the transient "shows all threads" flash is acceptable UX, or whether the axis should hold until currentUserName loads.
- Flashing intermediate state in UI while other things load has been called by the UX team in some other cases as not good. I don't know what "finding #1" is, exactly, but in general it's better to show some sort of empty/loading progress or linger on previous state depending on the situation. I don't know how realistic or disruptive the flashing is in this particular case, though.

3. New microcopy: verify the renamed labels ("All assignments", "All resolved statuses") read well in the live dropdowns and that the Spanish translations are natural.
- The English ones look good to me. I can't speak to the Spanish translations, but I think we're sticking with machine translations for now for strings that didn't have existing Spanish translations in PT9.

4. Narrow-layout rendering (finding #9): confirm the five-dropdown toolbar does not spill past three rows at the 300px webview min-width on long-label locales.
- I'm not sure why this is a reviewer task. Did you talk with anyone else already regarding the new split design? This diverges from PT9, so I'm a little concerned about it. I think it makes sense, but if PT9 is our central guide, I'm not sure if this is a good choice unless others are on board already.

5. Full npm run typecheck in a standard checkout (see typecheck caveat above).
- CI handles this

File: c-sharp/Projects/CommentThreadSelector.cs [hunk 1]
```diff
@@ -25,6 +25,15 @@ public class CommentThreadSelector
...
     public bool? IsRead { get; set; }
 
+    /// <summary>
+    /// If this is null (not set), then no resolved-status filtering will be applied. This filters on
+    /// the THREAD's status — the note-lifecycle sense of "resolved"
+    /// (<see cref="NoteStatus.Resolved"/>) — not on any conflict-specific state. <c>false</c> matches
+    /// threads whose status is anything other than Resolved (Todo, Done, or Unspecified), mirroring
+    /// PT9's <c>StatusFilter(StatusTypes.Unresolved)</c>.
+    /// </summary>
+    public bool? IsResolved { get; set; }
```
Comment (lyonsil) [inline c-sharp/Projects/CommentThreadSelector.cs:43]: **`isResolved` overlaps the existing `Status` field.** Both filter on `t.Status`, so `isResolved: true` is equivalent to `status: 'Resolved'`, and a contradictory combination (e.g. `status: 'Todo'` + `isResolved: true`) silently ANDs to zero results - the type system doesn't prevent it, only the new TSDoc caveat does. This extension never sets both, so it's latent, but it's a trap for future callers of the public selector. Worth considering whether a single set-valued / negatable `status` axis (unresolved = not-`Resolved`) could express the PT9 parity you need without a second overlapping field. Non-blocking.

File: extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  type: 'all',
+  assignment: 'all',
+};
+
+/**
+ * True when every comment-filter axis is at its `'all'` default — i.e. no axis is actively
+ * filtering. Does not consider the separate scope axis. Co-locates the axis-default knowledge with
+ * the axis definitions instead of duplicating the per-axis checks at call sites.
+ */
+export function areCommentFiltersAtDefault(filters: CommentFilters): boolean {
```
Comment (lyonsil) [inline extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts:106]: **Reuse `deepEqual` instead of the hand-written comparison.** `deepEqual` is exported from `platform-bible-utils` and already used for this exact "at default / has changes" idiom (`extensions/src/paratext-registration/src/internet-settings.web-view.tsx:94`). The body reduces to `return deepEqual(filters, DEFAULT_COMMENT_FILTERS);`. The manual field-by-field check also silently ignores any future 5th axis, which would then render `%no_comments%` ("No comments yet") instead of `%no_comments_match_filter%` when only that axis is active.

File: extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    filters.resolved === DEFAULT_COMMENT_FILTERS.resolved &&
+    filters.read === DEFAULT_COMMENT_FILTERS.read &&
+    filters.type === DEFAULT_COMMENT_FILTERS.type &&
+    filters.assignment === DEFAULT_COMMENT_FILTERS.assignment
+  );
+}
+
+// Paratext 9's literal "assigned to the whole team" token (UserFilter/CommentTags): threads assigned
+// to the team carry this exact `AssignedUser` value.
+export const TEAM_ASSIGNED_USER = 'Team';
```
Comment (lyonsil) [inline extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts:112]: **Third copy of the `'Team'` sentinel.** This literal is also hard-coded in `lib/platform-bible-react/src/components/advanced/comment-list/comment-list.utils.ts` (and documented in `comment-list.types.ts`), and is canonically `CommentThread.teamUser` in ParatextData. The value must equal `teamUser` for `assignedTo: 'Team'` to match anything in the provider (`t.AssignedUser == selector.AssignedTo`); if that token ever changes, "Assigned to team" silently returns zero threads and there are now three TS literals plus the C# constant to keep in sync. Consider a single shared exported constant as the source of truth.

File: extensions/src/legacy-comment-manager/src/comment-list.stories.tsx [hunk 4]
```diff
@@ -135,12 +137,15 @@ function filterThreads(
...
-    }
+    if (filters.resolved === 'unresolved' && thread.status === 'Resolved') return false;
+    if (filters.resolved === 'resolved' && thread.status !== 'Resolved') return false;
+    if (filters.read === 'unread' && thread.isRead) return false;
+    if (filters.read === 'read' && !thread.isRead) return false;
+    if (filters.type === 'conflicts' && thread.type !== 'Conflict') return false;
+    if (filters.type === 'comments' && thread.type !== 'Normal') return false;
+    if (filters.assignment === 'assigned-to-me' && thread.assignedUser !== currentUser)
+      return false;
+    if (filters.assignment === 'team' && thread.assignedUser !== TEAM_ASSIGNED_USER) return false;
```
Comment (lyonsil) [inline extensions/src/legacy-comment-manager/src/comment-list.stories.tsx:148]: **`filterThreads` is a parallel, hand-maintained implementation of the axis-to-match rules** already expressed by `buildCommentThreadSelector` + the C# `GetCommentThreads` predicate. Change an axis mapping in the model (e.g. the `comments -> Normal` or `isResolved` semantics) and this Storybook predicate silently keeps the old rule - the model's unit tests only cover `buildCommentThreadSelector`, so nothing flags the drift and the story then misrepresents shipped behavior. Consider exporting a single `threadMatchesFilters(thread, filters)` predicate from the model that the story reuses, so the axis semantics live in one place.

File: extensions/src/legacy-comment-manager/src/comment-list.web-view.tsx [hunk 5]
```diff
@@ -155,38 +153,16 @@ global.webViewComponent = function CommentListWebView({
...
-    ]),
+    useMemo<LegacyCommentThreadSelector>(
+      () =>
+        buildCommentThreadSelector({
+          filters,
+          scopeFilter,
+          scrRef: { book: scrRef.book, chapterNum: scrRef.chapterNum, verseNum: scrRef.verseNum },
+          currentUserName,
+        }),
+      [scrRef.book, scrRef.chapterNum, scrRef.verseNum, scopeFilter, filters, currentUserName],
```
Comment (lyonsil) [inline extensions/src/legacy-comment-manager/src/comment-list.web-view.tsx]: **Selector `useMemo` re-queries on every verse move even in the default view.** `scrRef.book/chapterNum/verseNum` are dependencies but only affect the built selector when `scopeFilter === 'current-chapter'`. In the default "all books" view, each verse move (e.g. synchronized scroll) rebuilds a new content-identical selector object; the data hook compares selectors by reference, so it tears down and re-establishes the subscription - re-running the C# `GetCommentThreads` and flashing the loading skeletons - with no change to the actual query. Pre-existing (these deps predate this PR; the diff only swapped `commentFilter -> filters` here), so non-blocking, but noting it since it's on the rewritten line.

File: extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+  // Read status (current user's per-thread read state)
+  if (filters.read === 'unread') selector.isRead = false;
+  else if (filters.read === 'read') selector.isRead = true;
+
+  // Note type
+  if (filters.type === 'conflicts') selector.type = 'Conflict';
+  else if (filters.type === 'comments') selector.type = 'Normal';
+
+  // Assignment
+  if (filters.assignment === 'assigned-to-me') selector.assignedTo = currentUserName;
```
Comment (lyonsil) [inline extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts]: **The assignment axis structurally can't express "unassigned."** `assignedTo: ''` already means "no filter" in the provider (`!string.IsNullOrEmpty`), so the field conflates the "no filter" sentinel with the real unassigned value (`""`, `CommentThread.unassignedUser`). PT9 surfaces a real "(Unassigned)" filter row; adding it here later can't use `assignedTo: ''`, so it'll need a sentinel token or a separate `isAssigned` field - the same overlapping-field trap as `isResolved` vs `status`. Flagging as an altitude consideration for when the dropped "Unassigned" option comes back. Non-blocking.

File: extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+export const resolvedFilterToLabelKey = {
+  all: '%comment_filter_resolved_all%',
+  unresolved: '%comment_filter_resolved_unresolved%',
+  resolved: '%comment_filter_resolved_resolved%',
+} as const;
+
+export type ResolvedFilter = keyof typeof resolvedFilterToLabelKey;
+
+export function isResolvedFilter(value: string): value is ResolvedFilter {
+  return value in resolvedFilterToLabelKey;
```
Comment (lyonsil) [inline extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts]: **Axis type-guards use `value in map`, which walks the prototype chain.** All five guards (`isScopeFilter`, `isResolvedFilter`, `isReadFilter`, `isTypeFilter`, `isAssignmentFilter`) accept inherited `Object.prototype` keys - `isResolvedFilter('toString')` returns `true`. Inert through `FilterDropdown` (`Object.keys()` yields own keys and Radix only emits rendered values), but unsound for any other caller: `labelKeys['toString']` would resolve to a function and blank the label. Prefer `Object.prototype.hasOwnProperty.call(map, value)`. (Pre-existing pattern in `isScopeFilter`, but this PR adds four more instances.)

**Review by lyonsil — COMMENTED** [review]
A few findings from a deeper review pass (altitude / reuse / efficiency). All non-blocking - flagging for consideration, not gating merge. AI-assisted review.

**Comment by rolfheij-sil** [comment]
🤖 Claude (drafted with Rolf — we worked through these responses together): On the design-divergence concern (your #4, and the P9-review question in #1): Rolf and I discussed this directly, and he confirmed he ran the orthogonal-axis split past Sebastian, who's on board with it for now. It's far from perfect, but it's a clear step up from the very limited preset filter we had before, and Sebastian was satisfied with this shape for the current pass — so this is a deliberate, agreed change rather than unreviewed drift.

I've also fixed the "Assigned to me" flash you flagged in #2 — the view now holds a loading state until the registration name resolves instead of briefly showing all threads.

I've replied inline on each of the other findings with what changed or why it's deferred.

File: c-sharp/Projects/CommentThreadSelector.cs (hunk 1 above)
Comment (rolfheij-sil) [reply c-sharp/Projects/CommentThreadSelector.cs:43]: 🤖 Claude: Fixed in `26b792f` — `GetCommentThreads` now throws `ArgumentException` if a selector sets both `Status` and `IsResolved`, so the contradictory combination fails fast instead of silently ANDing to zero. A single set-valued/negatable status axis would be a larger redesign touching the existing `status` callers, so I've left that as a possible follow-up.

File: extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts (hunk 2 above)
Comment (rolfheij-sil) [reply extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts:106]: 🤖 Claude: Fixed in `26b792f` — `areCommentFiltersAtDefault` is now just `deepEqual(filters, DEFAULT_COMMENT_FILTERS)`, which also stays correct if a 5th axis is added.

File: extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts (hunk 3 above)
Comment (rolfheij-sil) [reply extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts:112]: 🤖 Claude: Deferred. Within the extension it's already a single exported `TEAM_ASSIGNED_USER` (the story imports it), and I confirmed it equals `CommentThread.teamUser` in ParatextData. Unifying across the extension, platform-bible-react, and C# would need a shared constant C# can't consume — not worth it right now.

File: extensions/src/legacy-comment-manager/src/comment-list.stories.tsx (hunk 4 above)
Comment (rolfheij-sil) [reply extensions/src/legacy-comment-manager/src/comment-list.stories.tsx:148]: 🤖 Claude: Kept the Storybook predicate in sync by hand (it now also handles the new `unassigned` axis), but I didn't extract a shared `threadMatchesFilters` predicate — deferred as a follow-up if the drift risk bites again.

File: extensions/src/legacy-comment-manager/src/comment-list.web-view.tsx (hunk 5 above)
Comment (rolfheij-sil) [reply extensions/src/legacy-comment-manager/src/comment-list.web-view.tsx]: 🤖 Claude: Fixed in `26b792f` — the selector `useMemo` now freezes the `scrRef` inputs to constants unless the current-chapter scope is active, so a verse move in the all-books view no longer tears down and re-establishes the subscription.

File: extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts (hunk 6 above)
Comment (rolfheij-sil) [reply extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts]: 🤖 Claude: Fixed in `26b792f` — added the `unassigned` option back. The C# provider now treats `AssignedTo == null` as "any assignee" and `""` as unassigned (`CommentThread.unassignedUser`), so the sentinel collision is gone.

File: extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts (hunk 7 above)
Comment (rolfheij-sil) [reply extensions/src/legacy-comment-manager/src/comment-list-filters.model.ts]: 🤖 Claude: Fixed in `26b792f` — all five axis guards now use `Object.hasOwn(map, value)`, so inherited `Object.prototype` keys are rejected.

**Review by lyonsil — APPROVED** [review]
:lgtm:

> @lyonsil reviewed 12 files and all commit messages, made 1 comment, and resolved 7 discussions.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/2508#-:-Ox6UAY31zQQ5zAprYxM:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved.
